### PR TITLE
feat(registry): remove the documentation specifying insecure registry

### PR DESCRIPTION
**Pull request by kmala**

### CODE REVIEW

File: src/installing-workflow/system-requirements.md [hunk 1]
```diff
@@ -26,15 +26,6 @@ application footprint as well.
 Running smaller machines will likely result in increased system load and has been known to result in component failures
 and instability.
 
-## Docker Insecure Registry
```
Comment (slack) [inline src/installing-workflow/system-requirements.md]: We might want to keep the warning that "versions prior to 2.2 require '--insecure-registry' to function properly".   To keep our bases covered.

File: src/understanding-workflow/components.md [hunk 2]
```diff
@@ -94,8 +94,9 @@ configure minio to use persistent storage available in your environment.
 
 The registry component is a managed docker registry which holds application
 images generated from the builder component. Registry persists the Docker image
-iamges to either local storage (in development mode) or to object storage
-configured for the cluster.
+images to either local storage (in development mode) or to object storage
+configured for the cluster. The component also runs a [registry-proxy](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/registry/images) so that the
+registry service is exposed onto each node and available to docker daemon as localhost.
```
Comment (arschles) [inline src/understanding-workflow/components.md]: similar to the above comment (https://github.com/deis/workflow/pull/370/files#r70363312), might want to mention here that 2.1 doesn't do this

File: src/understanding-workflow/components.md [hunk 3]
```diff
@@ -94,8 +94,9 @@ configure minio to use persistent storage available in your environment.
 
 The registry component is a managed docker registry which holds application
 images generated from the builder component. Registry persists the Docker image
```
Comment (mboersma) [inline src/understanding-workflow/components.md]: This word "image" should be removed since the next line begins with it.

File: src/understanding-workflow/components.md [hunk 4]
```diff
@@ -94,8 +94,9 @@ configure minio to use persistent storage available in your environment.
...
-iamges to either local storage (in development mode) or to object storage
-configured for the cluster.
+images to either local storage (in development mode) or to object storage
+configured for the cluster. Starting with release V2.2.0 the component also runs a [registry-proxy](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/registry/images) daemon so that the
```
Comment (mboersma) [inline src/understanding-workflow/components.md]: The semver tag should start lowercase, and this should have a comma after that phrase. Maybe rephrase this part like this?
```
Starting with release v2.2.0, the component also runs a 
[proxy](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/registry/images)
daemon to expose the registry service on each node as `localhost`.

File: src/installing-workflow/system-requirements.md [hunk 5]
```diff
@@ -26,12 +26,9 @@ application footprint as well.
...
-
-The on-cluster Docker registry is not deployed with TLS enabled. As such, all Kubernetes worker nodes must have their
-Docker daemons configured to use an **insecure registry**. The configured subnet should encompass any private networks
-used by your worker nodes, including overlay networks.
+*** WARNING ***
```
Comment (bacongobbler) [inline src/installing-workflow/system-requirements.md]: I'd rather keep the old documentation, like this:

```
## Docker Insecure Registry

The on-cluster Docker registry is not deployed with TLS enabled. As such, versions prior to v2.2.0 must have all Docker daemons on the Kubernetes worker nodes configured to use an **insecure registry**. The configured subnet should encompass any private networks used by your worker nodes, including overlay networks.
```

In the future, if you want to add a warning in the docs, use the [admonition syntax](https://www.google.ca/#q=admonition+syntax+mkdocs&gws_rd=cr) so we can come up with a consistent style for these warnings.

```
!!! warning

    Versions prior to v2.2 require '--insecure-registry` to function properly.
```

File: src/understanding-workflow/components.md [hunk 6]
```diff
@@ -93,9 +93,11 @@ configure minio to use persistent storage available in your environment.
...
 
 The registry component is a managed docker registry which holds application
-images generated from the builder component. Registry persists the Docker image
-iamges to either local storage (in development mode) or to object storage
-configured for the cluster.
+images generated from the builder component. Registry persists the Docker
+images to either local storage (in development mode) or to object storage
+configured for the cluster. Starting with release v2.2.0, the component also runs a
+[proxy](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/registry/images)
+daemon to expose the registry service as `localhost` on each node.
```
Comment (bacongobbler) [inline src/understanding-workflow/components.md]: Can you elaborate more on how the proxy works and how this is a benefit over the old approach?

File: src/installing-workflow/system-requirements.md [hunk 7]
```diff
@@ -26,12 +26,9 @@ application footprint as well.
...
 and instability.
 
-## Docker Insecure Registry
-
-The on-cluster Docker registry is not deployed with TLS enabled. As such, all Kubernetes worker nodes must have their
-Docker daemons configured to use an **insecure registry**. The configured subnet should encompass any private networks
-used by your worker nodes, including overlay networks.
+!!! warning
 
+Versions prior to 2.2 require '--insecure-registry' to function properly.
```
Comment (bacongobbler) [inline src/installing-workflow/system-requirements.md]: This syntax won't work. You need to add 4 spaces, and then an empty newline character to signify the end of the admonition.

```
!!! warning

    Versions prior to 2.2 require '--insecure-registry' to function properly.

Depending on your...
```

File: src/installing-workflow/system-requirements.md [hunk 8]
```diff
@@ -26,12 +26,9 @@ application footprint as well.
...
 and instability.
 
-## Docker Insecure Registry
-
-The on-cluster Docker registry is not deployed with TLS enabled. As such, all Kubernetes worker nodes must have their
-Docker daemons configured to use an **insecure registry**. The configured subnet should encompass any private networks
-used by your worker nodes, including overlay networks.
+!!! warning
 
+    Versions prior to 2.2 require '--insecure-registry' to function properly.
```
Comment (bacongobbler) [inline src/installing-workflow/system-requirements.md]: This needs an extra newline character to signify the end of the admonition and the beginning of the paragraph, or else the next line will be considered part of the admonition.

Comment (kmala) [inline src/installing-workflow/system-requirements.md]: the next line also need to be part of admonition as it says how to set the insecure registry

Comment (bacongobbler) [inline src/installing-workflow/system-requirements.md]: Okay, then let's make it all one continuous paragraph.

Comment (kmala) [inline src/installing-workflow/system-requirements.md]: changed.

File: src/installing-workflow/system-requirements.md [hunk 9]
```diff
@@ -26,14 +26,10 @@ application footprint as well.
...
-## Docker Insecure Registry
+!!! warning
 
-The on-cluster Docker registry is not deployed with TLS enabled. As such, all Kubernetes worker nodes must have their
-Docker daemons configured to use an **insecure registry**. The configured subnet should encompass any private networks
-used by your worker nodes, including overlay networks.
-
-Depending on your Kubernetes and Docker configuration, setting `EXTRA_DOCKER_OPTS="--insecure-registry=10.0.0.0/8"` may
-be sufficient.
+    Versions prior to 2.2 require '--insecure-registry' to function properly.Depending on your Kubernetes and Docker configuration,
```
Comment (bacongobbler) [inline src/installing-workflow/system-requirements.md]: Needs whitespace after the period.

```
to function properly. Depending
```

**Comment by bacongobbler** [comment]
Can happily say that this LGTM. :shipit:!
